server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -640,26 +640,25 @@ private class ScanServerZKCleaner implements Runnable {
     public void run() {
 
       final ZooReaderWriter zrw = getContext().getZooReaderWriter();
-      final String sserverZNodePath = getContext().getZooKeeperRoot() + Constants.ZSSERVERS;
 
       while (stillManager()) {
         try {
-          for (String sserverClientAddress : zrw.getChildren(sserverZNodePath)) {
+          Set<ServiceLockPath> scanServerPaths =
+              getContext().getServerPaths().getScanServer(Optional.empty(), Optional.empty());
+          for (ServiceLockPath path : scanServerPaths) {
 
-            final String sServerZPath = sserverZNodePath + "/" + sserverClientAddress;
-            final var zLockPath = ServiceLock.path(sServerZPath);
             ZcStat stat = new ZcStat();
             Optional<ServiceLockData> lockData =
-                ServiceLock.getLockData(getContext().getZooCache(), zLockPath, stat);
+                ServiceLock.getLockData(getContext().getZooCache(), path, stat);
 
             if (lockData.isEmpty()) {
               try {
-                log.debug("Deleting empty ScanServer ZK node {}", sServerZPath);
-                zrw.delete(sServerZPath);
+                log.debug("Deleting empty ScanServer ZK node {}", path);
```

Comment:
This code was cleaning up empty scan server lock nodes, probably will not work now.  Wondering if we should have a third argument on methods like `getContext().getServerPaths().getScanServer(`  that specifies if the paths must have locks beneath them or not.  That would make each use consider if that is important or not.



Not something for this PR, we probably need an IT that checks that empty server nodes eventually go away in ZK.  I will open an issue about that.  Seems like it would be good to ensure that is being cleaned up for more dynamic envs like K8s.

Reply:
Opened #4882 and #4883 about this empty cleanup code.